zio-http/src/test/scala/zhttp/service/ClientSpec.scala

```diff
@@ -1,16 +1,65 @@
 package zhttp.service
 
-import zhttp.internal.HttpRunnableSpec
-import zhttp.service.client.ClientSSLHandler.ClientSSLOptions
-import zio.test.Assertion.anything
+import zhttp.http._
+import zhttp.internal.{AppCollection, HttpRunnableSpec}
+import zhttp.service.server._
+import zio.ZIO
+import zio.duration.durationInt
+import zio.test.Assertion.{anything, containsString, isEmpty, isNonEmpty}
+import zio.test.TestAspect.{flaky, sequential, timeout}
 import zio.test.assertM
 
 object ClientSpec extends HttpRunnableSpec(8082) {
-  val env           = ChannelFactory.auto ++ EventLoopGroup.auto()
-  override def spec = suite("Client")(
+
+  override def spec = {
+    suiteM("Client") {
+      app.as(List(clientSpec)).useNow
+    }.provideCustomLayerShared(env) @@ timeout(30 seconds) @@ sequential
+  }
+
+  def clientSpec = suite("Client")(
     testM("respond Ok") {
-      val actual = Client.request("http://api.github.com/users/zio/repos", ClientSSLOptions.DefaultSSL)
+      val actual = request(!! / "users" / "zio" / "repos")
       assertM(actual)(anything)
-    },
-  ).provideCustomLayer(env)
+    } +
+      testM("non empty content") {
+        val actual          = request(!! / "users" / "zio")
+        val responseContent = actual.flatMap(_.getBody)
+        assertM(responseContent)(isNonEmpty)
+      } +
+      testM("POST request expect non empty response content") {
+        val headers         = Headers.userAgent("zio-http test")
+        val response        = request(!! / "users" / "zio", method = Method.POST, headers = headers, content = "test")
+        val responseContent = response.flatMap(_.getBody)
+        assertM(responseContent)(isNonEmpty)
+      } +
+      testM("empty content") {
+        val actual          = request(!! / "users" / "zio" / "repos")
+        val responseContent = actual.flatMap(_.getBody)
+        assertM(responseContent)(isEmpty)
+      } +
+      testM("text content") {
+        val actual          = request(!! / "users" / "zio")
+        val responseContent = actual.flatMap(_.getBodyAsString)
+        assertM(responseContent)(containsString("user"))
+      },
+  ).provideCustomLayer(env) @@ flaky @@ timeout(5 second)
+
+  val env = EventLoopGroup.nio() ++ ChannelFactory.nio ++ ServerChannelFactory.nio ++ AppCollection.live
+
+  val clientApp: HttpApp[Any, Nothing] = Http.collectM[Request] {
```

Comment:
Let's keep the `app` inside each test. This is exactly what we are doing in `ServerSpec`.